### ISSUE TITLE
arrow-parens should not warn with flowtype annotation

### ISSUE THREAD

**Issue by Swatinem** [gajus/eslint-plugin-flowtype#129]
I have enabled the `arrow-parens` eslint rule to not allow parens around single param arrow functions. But this interacts poorly with flowtype annotations:

```js
type A = {};
const a1 = (a: A) => a; // gives a `arrow-parens` error
const a2 = a: A => a; // is a parse error
```

**Comment by danharper** [comment]
I can double check tonight, but I think this is either fixed in the latest version of ESLint (as they support basic type annotations now), or might have an alternative rule in eslint-plugin-babel.

**Comment by Swatinem** [comment]
I wasn’t using `eslint-plugin-babel`. Thanks for pointing me into the right direction, and sorry for the inconvenience.
